### PR TITLE
Fix use of a mismatching unicode path extra field in zip unarchiving

### PR DESCRIPTION
**Pull request by cwalther**
The zip spec ([§4.6.9](https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT)) says that a unicode path extra field is only to be used when its CRC matches the current file name. Using it unconditionally, as introduced in a080e0a06b67349063087556dccc0a06dff3b90d, is wrong. Commons.compress already does it right internally, there is no need to do anything in plexus-archiver.

These commits add some tests for the misbehavior using a specially crafted zip file (two variants supplied for completeness, although only one is useful for the test, see commit message) as well as a fix.

Originally the bug affected archive extraction, causing files to be extracted at wrong paths. This changed, probably unintentionally, with 2aec2bafd50f79fa86798d03a806c15786418358 (released in 4.2.0); since then only direct use of FileInfo has been affected, e.g. when using file selectors. The tests cover both situations.

For a real-world example where this caused problems, see https://repo1.maven.org/maven2/net/java/dev/jna/jna/5.6.0/jna-5.6.0.jar: It includes such a stale unicode path extra field for file `com/sun/jna/linux-x86-64/libjnidispatch.so` with contents `libjnidispatch.so`, meaning that plexus-archiver, unlike standards-compliant unarchivers, would extract that file in the root of the archive instead of the proper subfolder. This then caused https://download.eclipse.org/releases/2021-03/202103171000/plugins/com.sun.jna_5.6.0.v20200716-0148.jar (shipped with Eclipse 2021-03, built [here](https://git.eclipse.org/c/orbit/orbit-recipes.git/tree/jna/com.sun.jna_5.6.0)) to include it in that incorrect location, rendering it inoperable on Linux x86-64.

### CODE REVIEW

**Comment by plamentotev** [comment]
Thanks for the contribution and the detailed explanation. Would be great if if you can add `README.md` file together with the specially crafted files with instruction how they are crafted so they can be easily recreated.

File: src/test/java/org/codehaus/plexus/archiver/zip/PlexusIoZipFileResourceCollectionTest.java [hunk 1]
```diff
@@ -122,4 +123,21 @@ public void testSymlinkEntries()
...
+        PlexusIoZipFileResourceCollection prc = new PlexusIoZipFileResourceCollection();
+        prc.setFile( getTestFile( "src/test/resources/unicodePathExtra/efsclear.zip" ) );
+        Set<String> names = new HashSet<>();
+        final Iterator<PlexusIoResource> entries = prc.getEntries();
+        while ( entries.hasNext() )
+        {
+            final PlexusIoResource next = entries.next();
+            names.add(next.getName());
+        }
+        assertEquals( "should use good extra fields but not bad ones",
```
Comment (plamentotev) [inline src/test/java/org/codehaus/plexus/archiver/zip/PlexusIoZipFileResourceCollectionTest.java:138]: A comment here may make it more clear why the extra fields are bad.

**Comment by cwalther** [comment]
Both done. I also added the Java code I used to generate the test files.

**Comment by plamentotev** [comment]
Thanks a lot. Great work.

**Comment by cwalther** [comment]
Thanks!

**Pull request merged**